runtime-sdk/src/modules/consensus/mod.rs

```diff
@@ -152,6 +154,11 @@ impl API for Module {
     ) -> Result<(), Error> {
         Self::ensure_consensus_denomination(ctx, amount.denomination())?;
 
+        if ctx.is_check_only() {
+            Core::add_weight(ctx, CheckTxWeight::ConsensusMessages, 1)?;
```

Comment:
Should this weight be added in any case not just during checks? Then we could reject transactions in case it exceeds the per-batch maximum values (because a bad proposer could still include such transactions)?

Reply:
Hm yeah maybe? Although we already do have custom checks for messages and batch gas. Maybe we could remove those checks and generalize it into weight limits checks?

Reply:
Yeah I think this makes sense.

Reply:
I changed so that weights are added in all cases here, but i didn't add the per-batch weight checks. I have an initial implementation of the batch-limit checks, but I like it less than the current specialized gas accounting and message length checks. Maybe i'd suggest deferring on that until we add any more "weights" and just opening an issue for now.

I think that while "messages" and "gas" are the only "weights" that are used it doesn't make to much sense generalizing this. Manily since:
- gas accounting does a bit more than just checking batch weight limits (also tracks that the transaction amount spent is within the fee.gas_amount limit), so we have a special case here
- messages are quite first-class-y in the current code (e.g. both `Context` and `Dispatcher` are aware of messages), so i think it makes sense having the emitted message limits enforced the way it's done currently

Reply:
Ah right, message limits are also enforced, ok agreed no need for a generalized mechanism.

Reply:
Ah right, message limits are also enforced, ok agreed no need for a generalized mechanism.